src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py

```diff
@@ -1005,6 +1005,102 @@ def test_vm_disk_max_shares_etc(self, resource_group):
             self.check('maxShares', 1)
         ])
 
+    @ResourceGroupPreparer(name_prefix='cli_test_create_disk_from_diff_gallery_image_version_', location='westus')
+    def test_create_disk_from_diff_gallery_image_version(self):
+        self.kwargs.update({
+            'location': 'westus',
+            'vm': 'vm1',
+            'gallery1': self.create_random_name('gallery1', 16),
+            'gallery2': self.create_random_name('gallery2', 16),
+            'image1': 'image1',
+            'image2': 'image2',
+            'version': '1.1.2',
+            'disk1': 'disk1',
+            'disk2': 'disk2',
+            'disk3': 'disk3',
+            'subId': '0b1f6471-1bf0-4dda-aec3-cb9272f09590',
+            'tenantId': '54826b22-38d6-4fb2-bad9-b7b93a3e9c5a'
+        })
+
+        self.kwargs['vm_id'] = \
+            self.cmd('vm create -g {rg} -n {vm} --image ubuntults --data-disk-sizes-gb 10 --admin-username clitest1 '
+                     '--generate-ssh-key --nsg-rule NONE').get_output_in_json()['id']
+        time.sleep(70)
+
+        self.cmd('sig create -g {rg} --gallery-name {gallery1} --permissions community --publisher-uri publisher1 '
+                 '--publisher-email test@microsoft.com --eula eula1 --public-name-prefix name1')
+        self.cmd('sig share enable-community -g {rg} -r {gallery1}')
+
+        self.cmd('sig image-definition create -g {rg} --gallery-name {gallery1} --gallery-image-definition {image1} '
+                 '--os-type linux --os-state Specialized -p publisher1 -f offer1 -s sku1')
+        self.cmd('sig image-version create -g {rg} --gallery-name {gallery1} --gallery-image-definition {image1} '
+                 '--gallery-image-version {version} --virtual-machine {vm_id}')
+
+        public_name = self.cmd('sig show --gallery-name {gallery1} --resource-group {rg} --select Permissions', checks=[
+            self.check('provisioningState', 'Succeeded'),
+            self.check('length(sharingProfile.communityGalleryInfo.publicNames)', '1')
+        ]).get_output_in_json()['sharingProfile']['communityGalleryInfo']['publicNames'][0]
+        community_gallery_image_version = '/CommunityGalleries/' + public_name + '/Images/' + self.kwargs['image1'] \
+                                          + '/Versions/' + self.kwargs['version']
```

Comment:
After feature #21423 is completed, `community_gallery_image_version` can be obtained from the `sig image-version show-community` command instead